src/table/components/footer/PageOptions.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+interface PageOptionsProps {
+  totalPages: number;
+  page: number;
+}
+
+export const MAX_PAGE_OPTIONS_LENGTH = 1000;
+
+export const ELLIPSIS = '…';
+
+const PageOptions = ({ totalPages, page }: PageOptionsProps) => {
+  const pageOptionsLength = Math.min(totalPages, MAX_PAGE_OPTIONS_LENGTH);
+  const tooManyPages = pageOptionsLength < totalPages;
+  const endPageIndex =
+    tooManyPages && page > MAX_PAGE_OPTIONS_LENGTH / 2
```

Comment:
this is checked twice, could maybe be called something, might improve readability